[sil_tchad] simplify long presses

jeffheath-sil opened the pull request:
[sil_tchad] Simplify touch long presses, add Andika Afr font

keyman-server commented:
This pull request is from an external repo and will not automatically be built. The build must still be passed before it can be merged. Ask one of the team members to make a manual build of this PR.

DavidLRowe commented:
You eliminated a number of longpress keys from `a`, `e`, `i`, `o` and `u`. Did you want to eliminate the corresponding longpress keys from the shift layer (that is for `A`, `E`, `I`, `O` and `U`)?

On the shift layer of the touch layout, typing `Q`, `W` or `E` (but not the longpress keys on `E`) takes you back to the default layer, but typing any other key (including longpress keys) will leave you on shift layer. Is that intentional?

On the On-Screen Keyboard (OSK), the backslash key (rightmost key on second row) is blank, but produces the `\` character. It seems that this character should be added to that key on the OSK.

FYI: In the .kmn file, the &ETHNOLOGUECODE store is deprecated. It doesn't hurt (at least not yet) to have it in the .kmn file, but I don't believe it does anything anymore. A lot of people have commented it out (leaving it as a bit of documentation), but others have deleted it so it no longer needs to be kept up-to-date. (Up to you whether or not to make any changes.)

There's a welcome-fra.htm file included in the welcome folder, but it's not clear to me how that is invoked from the welcome.htm file.

jeffheath-sil commented:
Hi David,

 

These are good catches. Thank you!

 

Are you working on it right now? Because my Google Drive folder seems to have someone else accessing it… Would be good to coordinate.

 

Jeff

 

From: DavidLRowe <notifications@github.com> 
Sent: Friday, February 19, 2021 12:07 AM
To: keymanapp/keyboards <keyboards@noreply.github.com>
Cc: Jeff Heath <jeff_heath@sil.org>; Author <author@noreply.github.com>
Subject: Re: [keymanapp/keyboards] [sil_tchad] simplify long presses (#1464)

 

You eliminated a number of longpress keys from a, e, i, o and u. Did you want to eliminate the corresponding longpress keys from the shift layer (that is for A, E, I, O and U)?

On the shift layer of the touch layout, typing Q, W or E (but not the longpress keys on E) takes you back to the default layer, but typing any other key (including longpress keys) will leave you on shift layer. Is that intentional?

On the On-Screen Keyboard (OSK), the backslash key (rightmost key on second row) is blank, but produces the \ character. It seems that this character should be added to that key on the OSK.

FYI: In the .kmn file, the &ETHNOLOGUECODE store is deprecated. It doesn't hurt (at least not yet) to have it in the .kmn file, but I don't believe it does anything anymore. A lot of people have commented it out (leaving it as a bit of documentation), but others have deleted it so it no longer needs to be kept up-to-date. (Up to you whether or not to make any changes.)

There's a welcome-fra.htm file included in the welcome folder, but it's not clear to me how that is invoked from the welcome.htm file.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub <https://github.com/keymanapp/keyboards/pull/1464#issuecomment-781694235> , or unsubscribe <https://github.com/notifications/unsubscribe-auth/ADSC3OPUL5NZVD3DXDBIPBTS7WMS7ANCNFSM4X26XJOQ> .  <https://github.com/notifications/beacon/ADSC3OKGEDNDXU7Q37L7H5TS7WMS7A5CNFSM4X26XJO2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOF2L3KGY.gif>

jeffheath-sil commented:
Hi David,

 

Comments below…

 

From: DavidLRowe <notifications@github.com> 
Sent: Friday, February 19, 2021 12:07 AM
To: keymanapp/keyboards <keyboards@noreply.github.com>
Cc: Jeff Heath <jeff_heath@sil.org>; Author <author@noreply.github.com>
Subject: Re: [keymanapp/keyboards] [sil_tchad] simplify long presses (#1464)

 

You eliminated a number of longpress keys from a, e, i, o and u. Did you want to eliminate the corresponding longpress keys from the shift layer (that is for A, E, I, O and U)?

On the shift layer of the touch layout, typing Q, W or E (but not the longpress keys on E) takes you back to the default layer, but typing any other key (including longpress keys) will leave you on shift layer. Is that intentional?

JH> These are good observations, and I have fixed up the keyboard now so that the shift layer is handled better. The original changes were made months ago, and obviously I got interrupted in the middle of making the changes (and didn’t finish the work of the shift layer), so I’m glad you caught those problems.

On the On-Screen Keyboard (OSK), the backslash key (rightmost key on second row) is blank, but produces the \ character. It seems that this character should be added to that key on the OSK.

JH> Done.

FYI: In the .kmn file, the &ETHNOLOGUECODE store is deprecated. It doesn't hurt (at least not yet) to have it in the .kmn file, but I don't believe it does anything anymore. A lot of people have commented it out (leaving it as a bit of documentation), but others have deleted it so it no longer needs to be kept up-to-date. (Up to you whether or not to make any changes.)

JH> I deleted that line.

There's a welcome-fra.htm file included in the welcome folder, but it's not clear to me how that is invoked from the welcome.htm file.

JH> At some point in the past, the Windows keyboard installation was able to pull that up, but maybe that’s not possible anymore. Let me know if we should remove it, but I’m tempted to leave it in for now, just in case…

JH> OK, I believe I updated the pull request with these changes, but the process was a little different – I didn’t have to create the pull request again. If you don’t have all the files that you need, you can still access the Jeff-KeymanDev folder on Google Drive.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub <https://github.com/keymanapp/keyboards/pull/1464#issuecomment-781694235> , or unsubscribe <https://github.com/notifications/unsubscribe-auth/ADSC3OPUL5NZVD3DXDBIPBTS7WMS7ANCNFSM4X26XJOQ> .  <https://github.com/notifications/beacon/ADSC3OKGEDNDXU7Q37L7H5TS7WMS7A5CNFSM4X26XJO2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOF2L3KGY.gif>

DavidLRowe commented:
For the welcome-fra.htm file: 

I think you could include `<a href="welcome-fra.htm">Version française</a>` in the welcome.htm file and link the two. You'd probably want a corresponding link to get back to the English version. However, I'm not sure how that approach would work with the sil_tchad.php help file. We encourage people to keep the welcome.htm (the help packaged with the keyboard) and the .php file (the help available online) synchronized.

Another option is to include text for both languages in the welcome.htm file. Most keyboards that do that list one language and then the other, but the tem_kdh keyboard puts them side-by-side (release/t/tem_kdh/source/welcome/welcome.htm).

A third option, if you want to keep the file with this keyboard but not link to it, would be to create an `extras` folder at the same level as `source` and move the file there. You would also remove it from the packaging list so that we're not including in the package a file that cannot be accessed.

DavidLRowe commented:
@jeffheath-sil Since you're making additional commits to this pull request, would you want to include your revised .keyboard_info file so that the font information is available for Keyman Web users?

jeffheath-sil commented:
@DavidLRowe Thanks for your patience on this. I left the welcome-fra files as is, because they seem to do the right thing. (When Keyman UI is in French, shows Lisezmoi, and welcome text in French.) The changes listed in the HISTORY file are a fairly complete list, I believe, of the changes made in this version 4.2. Let me know if there is anything else needed.

DavidLRowe commented:
I didn't test every key, but the only anomaly I found was on the shift layer, with the longpress on the period. If I select `'` I'd expect to get U+0027, but actually get U+2019. Compare the code used on the shift level (default K_QUOTE) with the code used on the default layer (U_0027). (Or if you intend the shift layer to produce U+2019, it would be best to change the character displayed in the longpress menu.)

Other than this, I think this is ready for approval (assuming it builds successfully, which will need Lorna or me to manually start the build.

DavidLRowe commented:
I see in the .kps file that both the .ttf and .woff are included. @LornaSIL Is that usual? recommended?

LornaSIL commented:
I don't think woff files can be installed or used through a Keyman package.

DavidLRowe commented:
@jeffheath-sil I think that it would be better to remove the .woff file from the list of files in the .kps file, and remove the file itself from the source folder. 

I suppose you could put the .woff file in an `extras` folder (at the same level as `source` and `build`), but there doesn't seem to be a way for the Keyman system to actually use that font.

jeffheath-sil commented:
@DavidLRowe Thanks for the input. The straight quote on the longpress period seemed to be working for me, but I changed it to be the same as that on the shift layer, so hopefully that works for you now. But it did point out to me a problem... In the desktop keyboard, the apostrophe key gives a 2019 (for curved apostrophe in French j’aime or English can’t) and if you type it again it gives you 02BC for the orthographic glottal. The touch keyboard was just giving the glottal straightaway, and it was very inconvenient to get the 2019. So I changed it now so that that the touch layout gives 2019 the first time, then changes it to 02BC if you tap again. I added some special code to do that - not quite sure if it’s the best syntax, but it seems to work. You might want to check it out.

And I removed the .woff file from the project.

DavidLRowe commented:
Sorry I wasn't clear. Previously the straight apostrophe on the default layer was working as expected, that is, the character displayed is U+0027 and U+0027 was produced, whereas on the shift layer U+0027 is displayed, but U+2019 is produced. My recommendation would be to change both default and shift layer to use `U_0027` (rather than `K_QUOTE` on `default` layer), that way the key will produce U+0027.

All other changes look fine.

jeffheath-sil commented:
@DavidLRowe OK, I've fixed that. Might be good now?

DavidLRowe commented:
@jeffheath-sil The updated keyboard (v4.2) should be available at https://keyman.com/keyboards/sil_tchad before too long. Right now it's still on v3.3.2.
